sweagent/agent/agents.py

```diff
@@ -341,7 +353,10 @@ def save_trajectory(
             "history": self.history,
             "info": info,
         }
-        log_path.write_text(json.dumps(log_dict, indent=2))
+        try:
+            log_path.write_text(json.dumps(log_dict, indent=2, default=vars))
```

Comment:
`default=vars` is necessary because the trajectories now contain Anthropic API objects. `default=vars` (for some reason) is necessary to stringify those into JSON.